plugins/inventory/foreman.py

```diff
@@ -238,10 +293,214 @@ def _get_hostvars(self, host, vars_prefix='', omitted_vars=()):
                 hostvars[vars_prefix + k] = v
         return hostvars
 
+    def to_safe(self, word):
+        '''Converts 'bad' characters in a string to underscores
+        so they can be used as Ansible groups
+        >>> ForemanInventory.to_safe("foo-bar baz")
+        'foo_barbaz'
+        '''
+        regex = r"[^A-Za-z0-9\_]"
+        return re.sub(regex, "_", word.replace(" ", ""))
+
+    def _fetch_params(self):
+        options = ("no", "yes")
+        params = dict()
+        self.want_location = self.get_option('report').get('want_location')
+        self.want_organization = self.get_option('report').get('want_organization')
+        self.want_IPv4 = self.get_option('report').get('want_ipv4')
+        self.want_IPv6 = self.get_option('report').get('want_ipv6')
+        self.want_facts = self.get_option('report').get('want_facts')
+        self.want_host_group = self.get_option('report').get('want_host_group')
+        self.want_hostcollections = self.get_option('report').get('want_hostcollections')
+        self.want_subnet = self.get_option('report').get('want_subnet')
+        self.want_subnet_v6 = self.get_option('report').get('want_subnet_v6')
+        self.want_smart_proxies = self.get_option('report').get('want_smart_proxies')
+        self.want_content_facet_attributes = self.get_option('report').get('want_content_facet_attributes')
+        self.want_host_params = self.get_option('report').get('want_host_params')
+        self.host_filters = self.get_option('host_filters')
+
+        params["Organization".encode('ascii')] = options[self.want_organization]
+        params["Location".encode('ascii')] = options[self.want_location]
+        params["IPv4".encode('ascii')] = options[self.want_IPv4]
+        params["IPv6".encode('ascii')] = options[self.want_IPv6]
+        params["Facts".encode('ascii')] = options[self.want_facts]
+        params["Host Group".encode('ascii')] = options[self.want_host_group]
+        params["Host Collections".encode('ascii')] = options[self.want_hostcollections]
+        params["Subnet".encode('ascii')] = options[self.want_subnet]
+        params["Subnet v6".encode('ascii')] = options[self.want_subnet_v6]
+        params["Smart Proxies".encode('ascii')] = options[self.want_smart_proxies]
+        params["Content Attributes".encode('ascii')] = options[self.want_content_facet_attributes]
+        params["Host Parameters".encode('ascii')] = options[self.want_host_params]
+        if self.host_filters:
+            params["Hosts".encode('ascii')] = self.host_filters
```

Comment:
why the `encode('ascii')` everywhere?

Reply:
Meant to revert that. I was having some unrelated issues with encoding while working on this and was trying some fixes. Removed these.